scene/resources/material.cpp

```diff
@@ -1932,8 +1938,18 @@ void fragment() {)";
 	ShaderData shader_data;
 	shader_data.shader = RS::get_singleton()->shader_create_from_code(code);
 	shader_data.users = 1;
-	shader_map[mk] = shader_data;
-	shader_rid = shader_data.shader;
+
+	MutexLock lock(shader_map_mutex);
+	if (ShaderData *v = shader_map.getptr(mk); v) {
+		// We raced and managed to sneak the same key in concurrently, so we'll destroy the one we just created,
+		// given we know it isn't used, and use the winner.
+		RS::get_singleton()->free(shader_data.shader);
```

Comment:
@clayjohn I had to remove this, as it caused a crash in the `List` destructor of this list, as it wasn't empty:

https://github.com/godotengine/godot/blob/324512e11c1b7663c3cf47bec6ddbe65c6b8db2b/servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.h#L291

Reply:
Makes sense